core/src/main/java/org/apache/calcite/sql/parser/SqlParserPos.java

```diff
@@ -125,6 +127,17 @@ public int getEndColumnNum() {
     return endColumnNumber;
   }
 
+  /** Mark this SqlParserPos as quoted and return itself. **/
+  public SqlParserPos setQuoted(boolean quoted) {
```

Comment:
@danny0405, is this a little too radical to add this method to `SqlParserPos`? Could we somehow move it to SqlIdentifier?

Reply:
And anyway it's better to keep the immutability of `SqlParserPos` or `SqlIdentifier`.

Reply:
@zhztheplayer I add the setQuoted in SqlIdentifier for the first commit, but Julian think SqlIdentifier is already a little complex and it's better to add in SqlParserPos, then we can quote just a position for composite identifier, e.g. a.`b`.c.

Reply:
Oh, I'm sorry for missing the JIRA discussion. I'm reading -- my initial thinking is that it sounds a little wired a "position" has opportunity to hold a "quote".

> For immutability, can you please give an example of the bad case of non-immutable SqlParserPos ?

I know making the `quoted` settable is not breaking current code. But maybe a possible bad case is, someone may found it easy to change quoting of a `SqlParserPos` programmatically to decide that whether to make an identifier a context variable or a column, this is out of our control.
And I can see you may not want to add a parameter to constructor of SqlParserPos. If we do such a workaround that adds "quote" to "position", maybe like what Julian has said, making a sub-class to do that?